### DISCUSSION TITLE
22000.258.31.0

### DISCUSSION THREAD

**Discussion by valinet** [valinet/ExplorerPatcher#166] [Announcements]
Tested on build: 22000.258.

#### New features

* Cortana button now opens the Widgets panel
* Ability to choose what happens when clicking the Network icon in the system tray
* Possibility to use the legacy clock flyout
* Possibility to use the legacy volume flyout
* Fixes to fully support the classic theme, with a functional taskbar, system tray, Explorer windows, working context menus; read more about this feature [here](https://github.com/valinet/ExplorerPatcher/discussions/101)

#### Feature enhancements

* Reorganized settings in the GUI
* Added option not to have an accelerator for the `Properties` menu entry in `Win`+`X` (#162)

#### Fixes

* Fixed an issue where the Windows 10 window switcher failed to display some windows (#161)
* Fixed an issue that prevented Start from opening again until Explorer was restarted after opening File Explorer via the Start menu Explorer icon (#145)
* Fixed patching in libvalinet
* Fixed GUI launch path; GUI now launches in an external process, survives Explorer restarts

#### Experimental

The application can now be registered as a shell extension. This will enable the Explorer related functionality to work in Open/Save file dialogs as well. This is especially useful for users wanting proper support of the classic theme in Windows 11.

Please note that this is experimental. For the moment, the preferred installation method remains dropping the DLL in `C:\Windows`. For interested users, I invite you to test this functionality and report your findings in the discussions board.

To enable this, put the 2 DLLs (`ExplorerPatcher.amd64.dll` and `ExplorerPatcher.IA-32.dll`) in a secure folder (for example, `C:\Program Files\ExplorerPatcher`). Then, in that folder, run this command: `regsvr32 ExplorerPatcher.amd64.dll`. After elevation, a message will display informing you of the operation outcome, and if it went well, Explorer will restart displaying the old taskbar.

To uninstall, run `regsvr32 /u ExplorerPatcher.amd64.dll` in the same folder and preferably reboot the computer to unload the DLLs from all applications. Then, the files can be deleted just fine.

Please consult the [README](https://github.com/valinet/ExplorerPatcher/blob/master/README.md) for more details.

A detailed change log is available [here](https://github.com/valinet/ExplorerPatcher/blob/master/CHANGELOG.md). 

<hr /><em>This discussion was created from the release <a href='https://github.com/valinet/ExplorerPatcher/releases/tag/22000.258.31.0'>22000.258.31.0</a>.</em>

**Comment by Gaurav-Original-ClassicShellTester** [comment]
Did something change in the Win+X upon right click behavior in this version? Not only does the menu appear farther up now than expected but it also sometimes selects Properties instead of Desktop. Earlier behavior was rock solid reliable for me. Also if I right click the hot corner, sometimes the menu even appears partially outside of the screen. I managed to take a screenshot when that happened.
![image](https://user-images.githubusercontent.com/46422834/138221788-59795d20-64c8-44b4-b761-3f30a3268c6c.png)

![image](https://user-images.githubusercontent.com/46422834/138222097-002c9623-7d6e-42d0-906a-1d108b90456a.png)

**Reply by valinet** [reply]
Well, not really. I haven't changed anything there. I have tested this repeatedly now, no issues here.

**Reply by Gaurav-Original-ClassicShellTester** [reply]
> Well, not really. I haven't changed anything there. I have tested this repeatedly now, no issues here.

Ohh I wonder if no one is facing it. I am on Windows 11 dev channel 22471.1000. I have disabled StartAllBack completely and exited Open Shell and 7+ Taskbar Tweaker (the 3 other apps that hook into Explorer) before reporting/testing but about 40-50% of the times, the Win+X menu (only when it's opened with right click in bottom left hot corner) doesn't render correctly in this release or opens empty/partially outside the screen. Here's another screenshot. It seems to be happening fairly regularly.

![image](https://user-images.githubusercontent.com/46422834/138344771-fa53706f-5fc4-4289-b834-65db3d87ac30.png)

Update: Ah finally found out why! Maybe my own stupidity. I did regsvr32 ExplorerPatcher.amd64.dll _in addition to_ having to the dll in C:\Windows. After unregistering it, it stopped happening. So we are only supposed to do one or the other?

**Comment by Gaurav-Original-ClassicShellTester** [comment]
Btw this version fixes the issues with SAB and is the nicest one yet so definitely overall a huge improvement. 😊

**Comment by kolunio82** [comment]
i have 1000.22000.258.0, how can i update to get this woking?

**Comment by nonKurau** [comment]
Maybe that's because I'm on win11 dev branch 22483.1000, but my explorer went into endlessly restarting loop with this release. Returned to safe and working 22000.258.30.6.

**Reply by valinet** [reply]
It would be nice to find out why. Try to enable the console and see what it says in there... "safe and working" - every new version is as safe as the one it replaces, if not safer, and new releases just try to improve on the feature set, stability and correctness of the older releases. If there is a problem, it would be good to try to isolate it, debug it and eventually fix it.

Thanks.

**Reply by nonKurau** [reply]
Just wanted to emphasize that it worked without problems for me before this release.

Console:

![1](https://user-images.githubusercontent.com/92923033/138276485-ed8c92c8-9e1d-4cb2-96b2-19602f8c36e8.png)

**Reply by valinet** [reply]
So, why is it not working then? Is the latest version still problematic or not?

**Reply by sisericm** [reply]
I can confirm having the same explorer restart issue the latest dxgi.dll (22000.258.31.3 ).
Version Info: 
Windows 11 Home
OS build	22000.258
Windows Feature Experience Pack 1000.22000.258.0 (updates disabled to avoid 282)
Open shell installed but disabled.

The system starts up fine but after opening & closing a window, explorer enters a restart loop (approx 1sec uptime before restarting). So far, only reverting to the previous release, which is wonderfully rock solid :) - (thank you), has proved a successful mitigation.

I will see what I kind find on the console and post any developments.
Thank you again for your work on this, Valinet. The patcher is a complete lifesaver. I would have reverted to 10, had I not found this tool.

**Reply by nonKurau** [reply]
And for me 22000.282.31.3 solved loop problem and all works fine now.

**Reply by sisericm** [reply]
@nonemo13 fantastic news. Thanks for the head's up.
So 282 symbol problem is resolved now aswell? Great stuff.
Many Thanks & Kudos to you @valinet

**Reply by awegawegw** [reply]
Same issue here. I revert my dev branch to an early built and still have the same problem, but the 22000.258.30.6. works perfectly (at least on the early built of dev branch)

**Comment by Anixx** [comment]
I have made a guide on how to use the Windows Classic theme:
https://github.com/valinet/ExplorerPatcher/discussions/167

Can you please link it from the release page instead of our development discussion?

**Reply by valinet** [reply]
It will be included in an updated README soon™.

**Reply by Anixx** [reply]
OK!

**Comment by Anixx** [comment]
Your install shortcuts worked very well. Why not to include them into the release?

**Reply by valinet** [reply]
Good idea. The README has to be updated as well. I was taking it slowly... Don't want to push everyone to using that, rather wait a bit and let only users that seek that test it, so we know it works well, what problems they encounter, along the way, if any. But yeah, I may add the shortcuts as well, even though they just do `regsvr32 ExplorerPatcher.amd64.dll` and `regsvr32 /u ExplorerPatcher.amd64.dll`. I don't want to confuse people though, maybe rename them to "Install as shell extension"... I am and will be thinking about it.

**Reply by Anixx** [reply]
In the version you sent me there are many dlls not included in the release. I wonder if they are needed?

**Reply by valinet** [reply]
Not really. They are renames of older versions, forgot to delete. What I sent you is a precursor to what is now available in releases. There are no secrets. And don’t wonder so much, just chill :)))

**Comment by thuyhoangnet** [comment]


**Reply by nonKurau** [reply]
This issue probably: https://github.com/valinet/ExplorerPatcher/issues/129

**Reply by valinet** [reply]
Fixed

**Comment by tim-elvidge** [comment]
Have tried a few methods for having the taskbar ungrouped and this is the best by far.
Over the weekend I'll install Windows 11 on my main PC now that Ryzen bugs have been fixed and look through your source code to unravel this marvel!!

**Reply by valinet** [reply]
Thank you. Also, please, do take a look, you're more than welcome.

**Reply by tim-elvidge** [reply]
I have now updated my main PC:-
Windows 11 Enterprise - Domain Joined
Runing Hyper-V with two VMs running continuously (Windows Server 2022 Data Center and Windows 11 Pro)
Runing on a Ryzen 5900X with 32Gb ram
The upgrade went well from my own internal WSUS server and after patching to the latest preview build 22000.282
Windows Experience Windows Feature Experience Pack 1000.22000.282.0
Installed via dxgi.dll to c:\Windows and voila everything was tickety boo
Both Visual Studio 2019 and Visual Studion 2022 RC ran perfectly was able to build and deploy websites
All other apps performed as expected.
Over this weekend I'll dig into your code.

Thanks for the effort and time and creating a perfect Windows 11 experience for a person who has multiple copies of apps open and multiple browsers etc. etc.

I'll be putting $100 NZ dollars into your PayPal account as a token of my appreciation.

**Reply by valinet** [reply]
Thank you very much. I hope that we can keep this supported for as long as possible.

**Comment by Anixx** [comment]
In .4 the control panel applets still mostly appear with navbar and without clientedge:
![изображение](https://user-images.githubusercontent.com/1271736/138503878-53e2d977-b841-4897-b61e-c5e56fb01964.png)

**Reply by valinet** [reply]
When I will do an update there, it will be in the release notes. I haven't worked on that. Why post this for each dot release? It's not that big of a deal anyway. For me, the navbar never appears, for example. As for that client edge thing, just create an AHK script and call it a day. The way I do it is I put the extended style at window creation. Idk what Explorer is doing afterwards, haven't spent another day checking that. Please, compile the DLL yourself and test the multiple scenarios. It's not that hard. I have something else I am trying to fix at the moment, I can't be doing all the things at the same time. Figure some pattern that works and I will integrate that. It's very time consuming and I am kind of time stretched anyway, so it would be a real help. Thanks.

**Reply by Anixx** [reply]
For me it just happens all the time, with rare exceptions... As to making an AHK script, it would need besides adding the style also to resize the window forward and back so that the change be visible, and that often brings a bunch of other glitches...

**Reply by valinet** [reply]
Try with `dxgi.dll` as well, without installing as shell extension. See how it behaves like that...

**Reply by valinet** [reply]
That's be the only difference probably, I was testing like that at the moment.

**Reply by Anixx** [reply]
Yes, indeed, there is a difference. When the dll is put into the Windows folder, I usually do not see the navbar. 
Client Edge appears in folder "all tasks" (of the control panel):
![изображение](https://user-images.githubusercontent.com/1271736/138509449-739305e6-485b-4fbf-8741-be1b2cf0786b.png)
but does not in "Programs and components" and "Devices and printers". Still, there is no navbar, which is a plus compared to the regsvr32 method.
![изображение](https://user-images.githubusercontent.com/1271736/138509822-0fc3e727-868f-49e7-a9ef-324e205fbb63.png)

**Reply by Anixx** [reply]
When the All Tasks is started via shortcut to explorer.exe shell:::{ED7BA470-8E54-465E-825C-99712043E01C}, the clientedge is not added. If it is started as a folder named AllTasks.{ED7BA470-8E54-465E-825C-99712043E01C}, it is added. But the folder named so is displayed on the desktop without icon label in W11 (unlike W8).

**Reply by valinet** [reply]
Name it “All tasks.{…}”. It should display “All tasks”.

**Reply by Anixx** [reply]
I did it this way on Win8.1, but on Windows 11 it dos not show the name at all, only icon. I think, it is a bug.

**Reply by Anixx** [reply]
Well, so I just tried to copy all the DLLs into the Windows folder AND install them there. I do not know what is your opinion on such kind of installation, but so far it gives the best results (no navbar in control panel, plus context menus in dialogs).

**Reply by valinet** [reply]
Okay, so, first off, I know why the client edge was not working in "Control Panel", I fixed it.

[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7401064/dxgi.zip)

Secondly, I told you previously as well, without it being a shell extension, it won't inject other programs so you won't have context menus in there. That's what it mainly offers regarding that, plus hiding the nav bar and search bar in there as well in all cases. But indeed, it's been less tested than the classic method of dropping in `C:\Windows` which only injects Explorer.

Third, why are we talking here when we have a thread dedicated for this, so users looking for info on this specific topic have a central hub to get their info from.

**Reply by Anixx** [reply]
Thank you very much! All works as intended now!

As to installation method, I meant that using the both methods at the same time works the best: putting it into C:\Windows and registering it there. 
I am again sorry for writing is a wrong place, probably...

**Reply by valinet** [reply]
Yeah, because as it happens, when sued as shell extension, it gets loaded in "Control Panel" way after the nav bar was created, so of course it has not much to do from that point on. When injected as `dxgi.dll`, it loads as soon as possible, like, it cannot load sooner than that, so everything is untouched and the hooks hook the best.

I was about to suggest using both versions. That seems to be the way to go, although, there is still a bug which creates a nasty condition in File Explorer which eventually crashes some windows etc. I know the cause and it's been fixed already, it just triggers again in this condition. I will forward an update to you once I fix it here as well. So don't rely too much on that, wait to get this fix, please. You'll see, Control Panel windows crash after some time if left open if you run this dual dxgi-shell extension config.

**Reply by valinet** [reply]
![image](https://user-images.githubusercontent.com/6503598/138518081-4d8996e0-e9b7-4e72-985c-eb34ee2f050d.png)

Now it should be fix:
[Release.zip](https://github.com/valinet/ExplorerPatcher/files/7401161/Release.zip)

Indeed, drop all in `C:\Windows` for example and simply do `regsvr32 C:\Windows\ExplorerPatcher.amd64.dll`. Then, it should work fine. What still remains "unfixed":
* In 32-bit applications, some pick the nav bar setting from registry, so even if I hook too late, it's already taken care of; the rest will still display the nav bar
* FolderOptionsX is not working in file dialogs, so you won't get 3D borders there; yeah, someone would better port that over and integrate it into EP's infrastructure; they should have made it a shell extension from the get go, not a BHO, if you ask me, but whatever. Also, his answer for your question for adding FFLAGS fro desktop mode was lame... what's even "officially documented" these days anymore? So yeah, there's that as well.

But the good news is that on my part, hiding the search bar works when enabled, and also the context menus work, which I think they're a bit more important than client edge. I'd rather have the developer of FolderOptionsX come up with a solution for his DLL to get injected in all apps so it hooks Open/Save file dialogs as well. Compile a 32-bit DLL as well, etc. If not, I guess we could integrate that into EP at some point, idk. If someone volunteers to do that, it would be ideal.

**Reply by Anixx** [reply]
This works well for me!

**Reply by valinet** [reply]
As for the things FolderOptionsX does, if one knows a way to get the `IFolderView2` interface from, idk, the dialog window, then we can trigger the `SetFolderFlags` method with whatever we want... I don't have much experience in this area, unfortunately.

**Comment by kolunio82** [comment]
How can i get no rounded corners?

**Reply by kolunio82** [reply]
> 
> 
> #167

Which option? https://github.com/valinet/Win11DisableRoundedCorners/releases this don't work anymore.

**Reply by valinet** [reply]
> 
> 
> > #167
> 
> Which option? https://github.com/valinet/Win11DisableRoundedCorners/releases this don't work anymore.

Works just fine for me on 22000.282.

**Reply by kolunio82** [reply]
it shows unable to restore dwm. i have 1000.22000.282.0

**Reply by kolunio82** [reply]
There is a way to fix this?

**Comment by ghost** [comment]
Hi, I followed your steps but it didn't work out on the "never combine taskbar buttons" option I was looking for. However, executing these dll commands made my clock disappeared. :(  I uninstalled them (couldn't get the dxgi one out of Windows folder though) and rebooted, still doesn't show my taskbar as usual. Can you tell me how do I get it back?

**Reply by ghost** [reply]
Never mind. I went back to WIN10 already. :)

**Reply by BraINstinct0** [reply]
It seems like you clearly didn't follow the instructions, it is noted that deleting/moving the dxgi.dll file won't work when it is in use; you should rename it, restart Explorer.exe, and remove the file. If you don't have administrator-level access, well, this shouldn't be used from the beginning..

**Reply by ghost** [reply]
Clearly you're not helping. I even got myself extra help from the Internet to make sure I conducted it right. And I AM the administrator. So quit your condescending tone.
Who are you anyway? I wasn't asking for some random self-righteous prick's opinion.

**Reply by valinet** [reply]
> 
> 
> Clearly you're not helping. I even got myself extra help from the Internet to make sure I conducted it right. And I AM the administrator. So quit your condescending tone. Who are you anyway? I wasn't asking for some random self-righteous prick's opinion.

Hey, hey, keep it down. He wasn't wrong at all. He spared me saying it. The very answers to your original message are in right in the README. And you still make a scene, despite you having skipped over those. Or isn't it? Nevertheless, no one's under any obligation to help around here, you haven't got extra help from the Internet, maybe you got help, as help here is not a given, is a privilege. Everything here is provided as is, what comes on top, as I said, is a bonus.

**Reply by valinet** [reply]
And watch the language, please.

**Reply by BraINstinct0** [reply]
Sorry for being aggressive in the first place. I was trying to help(yes, I'm just another random user) but I do agree that I've sounded offensive. @kowabee hope the issue was solved, and I'm willing to help if any problem still exist.
@valinet will do.

**Comment by NandoZW** [comment]
Hi!, I noticed that some game launchers such as Epic Games and EA Launcher didn't run if the patch is installed. I don't know if I did something wrong, or it's just a bug.

**Reply by valinet** [reply]
Me neither. I don’t play games. Maybe their anti cheat protections detect this DLL being injected in Explorer and that’s their form of complaining. Idk for sure, through, as I said, I never installed/opened those programs. But yeah, if that’s the case, it’s definitely not a bug, but something by design.

**Reply by Mervinj82** [reply]
> 
> 
> Me neither. I don’t play games. Maybe their anti cheat protections detect this DLL being injected in Explorer and that’s their form of complaining. Idk for sure, through, as I said, I never installed/opened those programs. But yeah, if that’s the case, it’s definitely not a bug, but something by design.

Hi any luck with the means to replace the start button icon?

**Reply by valinet** [reply]
Hi. I did not have time to work on it. Some other user mentioned where in the theme file the image is, you could edit the theme and replace it there: https://github.com/valinet/ExplorerPatcher/discussions/219

**Comment by HuckFinn2021** [comment]
I'm using  Windows 11 Pro 21H2 Build 22000.282.  I downloaded and installed the 3 assets 
dxgi.dll 348 KB
ExplorerPatcher.amd64.dll 348 KB
ExplorerPatcher.IA-32.dll 135 KB
placed dxgi.dll in/windows, ran rundll32 C:\Windows\dxgi.dll,ZZRestartExplorer
placed 2 DLLs (ExplorerPatcher.amd64.dll and ExplorerPatcher.IA-32.dll) in a secure C:\Program Files\ExplorerPatcher) and ran regsvr32 /u ExplorerPatcher.amd64.dll.

I have tried for a day and 1/2 to get the system date/ time and sound , power, network icons to   show in the system tray (notification area) with no success,       

Can you advise how to get them to display.

![image](https://user-images.githubusercontent.com/93449934/139556958-7b0ffc23-a2bf-4bc5-9dc7-cf2fc799a3ed.png
[GitHub Picture.docx](https://github.com/valinet/ExplorerPatcher/files/7447531

/GitHub.Picture.docx)
)

**Reply by Mervinj82** [reply]
> 
> 
> I'm using Windows 11 Pro 21H2 Build 22000.282. I downloaded and installed the 3 assets dxgi.dll 348 KB ExplorerPatcher.amd64.dll 348 KB ExplorerPatcher.IA-32.dll 135 KB placed dxgi.dll in/windows, ran rundll32 C:\Windows\dxgi.dll,ZZRestartExplorer placed 2 DLLs (ExplorerPatcher.amd64.dll and ExplorerPatcher.IA-32.dll) in a secure C:\Program Files\ExplorerPatcher) and ran regsvr32 /u ExplorerPatcher.amd64.dll.
> 
> I have tried for a day and 1/2 to get the system date/ time and sound , power, network icons to show in the system tray (notification area) with no success,
> 
> Can you advise how to get them to display.
> 
> ![image](https://user-images.githubusercontent.com/93449934/139556958-7b0ffc23-a2bf-4bc5-9dc7-cf2fc799a3ed.png [GitHub Picture.docx](https://github.com/valinet/ExplorerPatcher/files/7447531
> 
> /GitHub.Picture.docx) )

I think you need to right click on the start button, choose properties and the properties window (or whatever it's called) will open. Select enable system tray icons and another window will pop up where you can select what to show in the system tray. By the way I think all this is in the read me file if I remember clearly.

**Reply by valinet** [reply]
Thanks @Mervinj82. It indeed is clearly written in the README.

I honestly do not understand what one could try so hard that it takes 1 day 1/2 and still fail… not to mention, why waste so much time instead of “wasting” 10 minutes to read the README through fully. Or at least ask from the get go, since it’s anyway the same thing, a day earlier or later, but at least you do not waste 1 1/2 days with… whatever you have tried.

**Comment by VV-SH** [comment]
I did not understand - is it necessary to remove dxgi.dll before registering ExplorerPatcher.amd64.dll?

**Reply by valinet** [reply]
No. But don’t bother with the shell extension unless you are ready for some rough edges here and there. The officially sanctioned install method is to drop the DLL in C:\Windows.

**Reply by VV-SH** [reply]
OK, TNX.
I tried to delete dxgi.dll and register explorerpatcher.amd64.dll, but received too many mistakes.
I will try not to delete dxgi.dll

**Reply by VV-SH** [reply]
Registered ExplorerPatcher.amd64.dll
Restarted Windows.
Desktop Viber is not loaded.
More precisely, the module is loaded into memory, but the interface is not displayed.
Skype - similarly.
Previously, Lock Screen was some kind of drawing, now Lock Screen is black.

**Reply by valinet** [reply]
Ok.
